common/ECSResourcesCheck/src/main/java/Comparer.java

```diff
@@ -0,0 +1,70 @@
+import org.json.JSONObject;
+public class Comparer {
+
+    public void compare(JSONObject jsonBase, String cluster,ECSResources ecs) {
+        Object[] arr = jsonBase.keySet().toArray();
+        JSONObject output = new JSONObject();
+        int key=jsonBase.keySet().toArray().length;
+        for (int i = 0; i < key; i++) {
+            System.out.println(arr[i].toString());
+            JSONObject base = jsonBase.getJSONObject(arr[i].toString());
+            JSONObject target = ecs.describeTskDef(arr[i].toString());
+            checkDifferences(target,base,cluster);
+            output.accumulate(arr[i].toString(),checkDifferences(target,base,cluster));
+        }
+        System.out.println(output);
+    }
+
+    public JSONObject checkDifferences (JSONObject target,JSONObject base,String cluster){
+        JSONObject output =new JSONObject();
+        if (target==null){
```

Comment:
Rather than testing for NULL of target here after invoking the checkDifferences function, why not check for it inside compare() right after calling ecs.describeTskDef(arr[i].toString()).  More efficient this way.

Reply:
Removed from this method -> moved to "compare" method